src/contexts/ToastsContext/Provider.tsx

```diff
@@ -30,22 +30,38 @@ export const ToastsProvider: React.FC = ({ children }) => {
     [setToasts],
   )
 
-  const toastError = (title: Toast['title'], description?: Toast['description']) => {
-    return toast({ title, description, type: toastTypes.DANGER })
-  }
-  const toastInfo = (title: Toast['title'], description?: Toast['description']) => {
-    return toast({ title, description, type: toastTypes.INFO })
-  }
-  const toastSuccess = (title: Toast['title'], description?: Toast['description']) => {
-    return toast({ title, description, type: toastTypes.SUCCESS })
-  }
-  const toastWarning = (title: Toast['title'], description?: Toast['description']) => {
-    return toast({ title, description, type: toastTypes.WARNING })
-  }
-  const clear = () => setToasts([])
-  const remove = (id: string) => {
+  const toastError = useCallback(
+    (title: Toast['title'], description?: Toast['description']) => {
+      return toast({ title, description, type: toastTypes.DANGER })
+    },
+    [toast],
+  )
+
+  const toastInfo = useCallback(
+    (title: Toast['title'], description?: Toast['description']) => {
+      return toast({ title, description, type: toastTypes.INFO })
+    },
+    [toast],
+  )
+
+  const toastSuccess = useCallback(
+    (title: Toast['title'], description?: Toast['description']) => {
+      return toast({ title, description, type: toastTypes.SUCCESS })
+    },
+    [toast],
+  )
+
+  const toastWarning = useCallback(
+    (title: Toast['title'], description?: Toast['description']) => {
+      return toast({ title, description, type: toastTypes.WARNING })
+    },
+    [toast],
+  )
+
+  const clear = useCallback(() => setToasts([]), [])
+  const remove = useCallback((id: string) => {
     setToasts((prevToasts) => prevToasts.filter((prevToast) => prevToast.id !== id))
-  }
+  }, [])
```

Comment:
This helps a lot when we have `toastError` or etc in useEffect deps. And this is the reason why we keep asking for request network change